### PR TITLE
Handle Docs Redirections

### PR DESCRIPTION
**Pull request by bryanculver**
# Closes: #DNE
# What's Changed

- [x] Review #4041
- [x] Add redirects for simple moved/renamed files
    - `git show --stat 46e2210009d5602eef436cfe90e74f4462e72058` filter for `nautobot/docs/{[before] => [after]}/[file].md`
- [x] Add redirects for deleted
    - `git show --stat 46e2210009d5602eef436cfe90e74f4462e72058` filter for deleted
    - `git diff ltm/1.6..develop -- '***.md' | grep "deleted file" -B 1`

The method of my madness has been to review #4041, put in redirects, and check build output for any errors, such as:

```
WARNING  -  Redirect target 'user-guide/administration/installation/ubuntu.md' does not exist!
```

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [x] Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Comment by itdependsnetworks** [comment]
nice solution!

**Comment by bryanculver** [comment]
Current page that 404s: https://docs.nautobot.com/projects/core/en/stable/plugins/development/

---
With PR:
```bash
➜ curl http://localhost:8001/projects/core/en/stable/plugins/development/

<!doctype html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>Redirecting...</title>
    <link rel="canonical" href="../../development/apps/">
    <meta name="robots" content="noindex">
    <script>var anchor=window.location.hash.substr(1);location.href="../../development/apps/"+(anchor?"#"+anchor:"")</script>
    <meta http-equiv="refresh" content="0; url=../../development/apps/">
</head>
<body>
Redirecting...
# ...
```
Redirecting to: http://localhost:8001/projects/core/en/stable/development/apps/ (or https://docs.nautobot.com/projects/core/en/stable/development/apps/)

**Comment by bryanculver** [comment]
> Current page that 404s: https://docs.nautobot.com/projects/core/en/stable/plugins/development/
> 
> With PR:
> 
> ```shell
> ➜ curl http://localhost:8001/projects/core/en/stable/plugins/development/
> 
> <!doctype html>
> <html lang="en">
> <head>
>     <meta charset="utf-8">
>     <title>Redirecting...</title>
>     <link rel="canonical" href="../../development/apps/">
>     <meta name="robots" content="noindex">
>     <script>var anchor=window.location.hash.substr(1);location.href="../../development/apps/"+(anchor?"#"+anchor:"")</script>
>     <meta http-equiv="refresh" content="0; url=../../development/apps/">
> </head>
> <body>
> Redirecting...
> # ...
> ```
> 
> Redirecting to: http://localhost:8001/projects/core/en/stable/development/apps/ (or https://docs.nautobot.com/projects/core/en/stable/development/apps/)

Also try it out yourself: https://docs-landing-page--4669.org.readthedocs.build/projects/core/en/4669/plugins/development/

File: mkdocs.yml [hunk 1]
```diff
@@ -102,6 +102,151 @@ plugins:
             heading_level: 1
             show_root_heading: true
             show_root_members_full_path: true
+  - "redirects":
+      redirect_maps:
+        'plugins/index.md': 'development/apps/index.md'
```
Comment (glennmatthews) [inline mkdocs.yml:107]: Might it make sense to alphabetize the keys here?

File: nautobot/docs/requirements.txt [hunk 2]
```diff
@@ -6,6 +6,7 @@ mkdocs-gen-files==0.5.0
 mkdocs-include-markdown-plugin==4.0.4
 mkdocs-material==9.1.18
 mkdocs-section-index==0.3.5
+mkdocs-redirects==1.2.1
```
Comment (glennmatthews) [inline nautobot/docs/requirements.txt:9]: ordering nit:

```suggestion
mkdocs-redirects==1.2.1
mkdocs-section-index==0.3.5
```

File: pyproject.toml [hunk 3]
```diff
@@ -179,6 +179,8 @@ mkdocs-include-markdown-plugin = "~4.0.4"
 mkdocs-material = "~9.1.18"
 # Automatically handle index pages for docs sections
 mkdocs-section-index = "~0.3.5"
+# Handle docs redirections
+mkdocs-redirects = "~1.2.1"
```
Comment (glennmatthews) [inline pyproject.toml:183]: Ordering nit:

```suggestion
# Handle docs redirections
mkdocs-redirects = "~1.2.1"
# Automatically handle index pages for docs sections
mkdocs-section-index = "~0.3.5"
```

File: changes/4669.fixed [hunk 4]
```diff
@@ -0,0 +1 @@
+Fixed broken links in docs from restructure project.
```
Comment (glennmatthews) [inline changes/4669.fixed:1]: Perhaps?

```suggestion
Added redirects from 1.x documentation paths to their 2.x equivalents to fix broken links/bookmarks.
```

**Review by glennmatthews — COMMENTED** [review]
Nice work!

**Review by glennmatthews — APPROVED** [review]
👏

**Pull request merged**